docs/formats/zst.md

```diff
@@ -190,128 +117,132 @@ independently of each other, similar to zng compression framing.
+of top-level Zed types that are present in the encoded input.
+To simplify terminology, we call a top-level Zed type a "super type",
+e.g., there are N unique super types encoded in the ZST object.
 
-This reassembly zng stream encodes 2*N+1 zng records, where N is equal to the number
-of top-level zng record types that are present in the encoded input.
-To simplify terminology, we call a top-level zng record type a "schema",
-e.g., there are N unique schemas encoded in the zst object.
+These N super types are defined by the first N values of the reassembly stream
+and are encoded as a null value of the indicated super type.
+A super type's integer position in this sequence defines its identifier
+encoded in the super column (defined below).  This identifier is called
+the super ID.
 
-Each of these N schemas gets defined as a record value, comprised
-of unset fields appearing in the same order
-as they were encountered in the original zng stream.
-In this way, a fresh zng type context can be created to read these N records and
-that type context will have precisely the same structure as the type context
-from the original zng stream.  This is guaranteed by the deterministic nature
-of embedded zng type definitions.
+> Change the first N values to type values instead of nulls?
 
-The next N+1 records contain reassembly information for each of the N schemas
-where each record is used to create column streams to reconstruct the original
-zng records.  The schemas do not overlap as columns from a record from any one
-schema are not intermixed with columns of another.
+The next N+1 records contain reassembly information for each of the N super types
+where each record defines the column streams needed to reconstruct the original
+Zed values.
 
 #### Segment Maps
 
-The foundation of column reconstruction is based on segment maps, which is simply
-a list of the segments from the data area that are concatenated to form the
-data for a column stream.
+The foundation of column reconstruction is based on _segment maps_.
+A segment map is a list of the segments from the data area that are
+concatenated to form the data for a column stream.
 
 Each segment map that appears within the reassembly records is represented
-with a zng array of records that represent seek ranges, i.e., specifically
-this zng type:
+with a Zed array of records that represent seek ranges conforming to this
+type signature:
 ```
-array[record[offset:uint64,length:uint32]]
+[{offset:uint64,length:uint32}]
 ```
 In the rest this document, we will refer to this type as `<segmap>` for
 shorthand and refer to the concept as a "segmap".
 
 > We use the type name "segmap" to emphasize that this information represents
-> a set of seek ranges where data stored and must be read from *rather than*
+> a set of byte ranges where data stored and must be read from *rather than*
 > the data itself.
 
-#### The Root Reassembly Record
+#### The Super Column
```

Comment:
The data section is just a bunch of arrays of data on disk.  By the logic here, you are saying all of the reassembly description should go in the data section because that's where the data is.  Anyway, this probably can be improved but I'm less concerned about ZST right now (compared to the other stuff) because it's not yet in wide use, has a version number, and we know it needs more work.  We can improve the description later.